rfcs/0001-buildpack-order.md

```diff
@@ -0,0 +1,60 @@
+# Proposed Buildpack Ordering in a Builder
+
+## Summary
+
+`pack inspect-builder gcr.io/paketo-buildpack/builder` should display a `Detection Order` that is deterministic and will never make it impossible for any buildpack to detect.
+
+## Motivation
+
+After a change to the builder detection order it became impossible for `php` apps to detect, because `nginx` and `httpd` were before `php` in the detect order. 
+The `httpd.conf` or `nginx.conf` included in `php` apps will cause the `nginx` or `http` buildpacks to detect `true` before `php` ever gets to run `detect`,
+making it impossible to access the `php` buildpack.
+
+
+## Detailed Explanation
+
+We propose the following rule to order buildpacks in the builder's `Detect Order`:
+
+- All meta buildpacks should be come before implementation buildpacks, and sort each group by buildpaco-id.
+
+- To prove this will always be valid: (1)
+    - Consider 2 buildpacks A and B.
+    - If any of buildpack B's order groups are a subset of any of buildpack A's order groups, then buildpack A must come before buildpack B in the builder order. 
+    - An implementation buildpack can be defined as an order group of itself.
+
+Thus, our exiting buildpacks would be ordered as follows:
+
+- paketo-buildpacks/dotnet-core
+- paketo-buildpacks/go
+- paketo-buildpacks/java
+- paketo-buildpacks/nodejs
+- paketo-buildpacks/php
+- paketo-community/python
+- paketo-community/ruby
+- paketo-buildpacks/httpd
+- paketo-buildpacks/nginx
+- paketo-buildpacks/procfile
```

Comment:
Procfile should be last, otherwise staticfile apps can't have a Procfile and detect

Reply:
I think it would only matter in the case we bundle `staticfile` and `procfile` into one order group. Right now `statcifile` doesn't know anything about Procfiles so people shouldn't include them in their app.

Reply:
Would we change the order later if we added Procfile support to the Staticfile group?

Reply:
Yeah that would be the idea, as it would fail the validator we dreamed up at that point. I do generally agree with you that we might want to just keep the old ordering though.